src/lib/features/feature-toggle/feature-toggle-store.ts

```diff
@@ -718,6 +725,36 @@ export default class FeatureToggleStore implements IFeatureToggleStore {
 
         return result?.potentially_stale ?? false;
     }
+
+    async setCreatedByUserId(batchSize: number): Promise<void> {
+        if (!this.flagResolver.isEnabled('createdByUserIdDataMigration')) {
+            return;
+        }
+        const toUpdate = await this.db(`${TABLE} as f`)
+            .joinRaw(`JOIN ${EVENTS_TABLE} AS ev ON ev.feature_name = f.name`)
+            .joinRaw(
+                `LEFT OUTER JOIN ${USERS_TABLE} AS u on ev.created_by = u.username OR ev.created_by = u.email`,
+            )
+            .joinRaw(
+                `LEFT OUTER JOIN ${API_TOKEN_TABLE} AS t on ev.created_by = t.username`,
+            )
+            .whereRaw(
+                `f.created_by_user_id IS null AND ev.type = 'feature-created'`,
```

Comment:
Maybe `f.created_by_user_id = -1337 AND f.created_by != 'system-user'`. I don't think this is as simple, but the idea to fix the other situation

Reply:
Thank you @gastonfournier! Yeah that is related to migration in `events` and there's a separate task for that! Thanks for the info, good help!👍